### PR TITLE
feat(apigateway): authorizationScopes for a Method

### PR DESCRIPTION
**Pull request by nirvana124**
## Commit Message
feat(apigateway): authorizationScopes for a Method (#6502)

Allowing to add authoriation scopes to api gateway
## End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-apigateway/lib/method.ts [hunk 1]
```diff
@@ -75,6 +75,12 @@ export interface MethodOptions {
    * The ID of the associated request validator.
    */
   readonly requestValidator?: IRequestValidator;
+
+  /**
+   * The authorizationScopes for the method
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/lib/method.ts]:
```suggestion
   * A list of authorization scopes configured on the method. The scopes are used with 
   * a COGNITO_USER_POOLS authorizer to authorize the method invocation.
   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-method.html#cfn-apigateway-method-authorizationscopes
```

File: packages/@aws-cdk/aws-apigateway/lib/method.ts [hunk 2]
```diff
@@ -75,6 +75,12 @@ export interface MethodOptions {
...
+   * @default none
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/lib/method.ts]:
```suggestion
   * @default - empty list
```

File: packages/@aws-cdk/aws-apigateway/lib/method.ts [hunk 3]
```diff
@@ -152,7 +158,8 @@ export class Method extends Resource {
       integration: this.renderIntegration(props.integration),
       methodResponses: this.renderMethodResponses(options.methodResponses),
       requestModels: this.renderRequestModels(options.requestModels),
-      requestValidatorId: options.requestValidator ? options.requestValidator.requestValidatorId : undefined
+      requestValidatorId: options.requestValidator ? options.requestValidator.requestValidatorId : undefined,
+      authorizationScopes: options.authorizationScopes || defaultMethodOptions.authorizationScopes
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/lib/method.ts]:
```suggestion
      authorizationScopes: options.authorizationScopes ?? defaultMethodOptions.authorizationScopes
```

File: packages/@aws-cdk/aws-apigateway/test/test.method.ts [hunk 4]
```diff
@@ -693,6 +693,57 @@ export = {
       });
     }, /Authorization type is set to NONE which is different from what is required by the authorizer/);
 
+    test.done();
+  },
+  'method has Auth Scopes'(test: Test) {
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/test/test.method.ts]: New line between test cases

```suggestion

  'method has Auth Scopes'(test: Test) {
```

File: packages/@aws-cdk/aws-apigateway/test/test.method.ts [hunk 5]
```diff
@@ -693,6 +693,57 @@ export = {
...
+
+    // THEN
+    expect(stack).to(haveResource('AWS::ApiGateway::Method', {
+      ApiKeyRequired: true,
+      AuthorizationScopes: ['AuthScope1', 'AuthScope2']
+    }));
+
+    test.done();
+  },
+  'use default Auth Scopes'(test: Test) {
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/test/test.method.ts]:
```suggestion

  'use default Auth Scopes'(test: Test) {
```

File: packages/@aws-cdk/aws-apigateway/test/test.method.ts [hunk 6]
```diff
@@ -693,6 +693,57 @@ export = {
...
+        operationName: 'defaultAuthScopes'
+      }
+    });
+
+    // THEN
+    expect(stack).to(haveResource('AWS::ApiGateway::Method', {
+      OperationName: 'defaultAuthScopes',
+      AuthorizationScopes: ['DefaultAuth']
+    }));
+
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/test/test.method.ts]: 2 test cases missing -

1. given no authorization scopes, verify that the property is absent. Use [`ABSENT` feature](https://github.com/aws/aws-cdk/tree/master/packages/%40aws-cdk/assert#check-existence-of-a-resource) of the assert library.

2. given authorization scope in method options and default method options, the one in method options is picked up.

Comment (nirvana124) [reply packages/@aws-cdk/aws-apigateway/test/test.method.ts]: Updated as per comments.

**Comment by nirvana124** [comment]
@nija-at All suggested changes are done.

File: packages/@aws-cdk/aws-apigateway/README.md [hunk 7]
```diff
@@ -678,6 +678,15 @@ properties and new resource types will not be available.
...
 
+## Configure AuthorizationScopes for a method
+AuthorizationScopes are used with COGNITO_USER_POOLS to authorize method invocation. More info about AuthorizationScopes can be found [here] https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-method.html#cfn-apigateway-method-authorizationscopes
+
+```ts
+books.addMethod('GET', new apigateway.HttpIntegration('http://amazon.com'), {
+  authorizationType: AuthorizationType.COGNITO,
+  authorizationScopes: ['Scope1','Scope2']
+});
+```
```
[the remark this answers is not in the document]
Comment (nirvana124) [reply packages/@aws-cdk/aws-apigateway/README.md]: @nija-at Changes done as mentioned.

**Pull request merged**